### ISSUE TITLE
Coverity vulnerabilities in alpine-librdkafka.so and centos6-librdkafka.so

### ISSUE THREAD

**Issue by bmgbeokwere** [edenhill/librdkafka#3290]
Read the FAQ first: https://github.com/edenhill/librdkafka/wiki/FAQ

Description
===========
We run our microservices through a binary analysis tool and both alpine-librdkafka.so and centos6-librdkafka.so look to have outdated references to openssl. Alpine looks to be referencing 1.0.2u while centos6 references 1.1.1i. Can we update these references to the latest versions?

In addition, is there a way to better manage these dependencies? It seems that we would lag on protection from know vulnerabilities if we require a new release to guard ourselves.

How to reproduce
================
Run through a binary analysis tool.

**IMPORTANT**: Always try to reproduce the issue on the latest released version (see https://github.com/edenhill/librdkafka/releases), if it can't be reproduced on the latest version the issue has been fixed.

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:

 - [x] librdkafka version (release number or git tag): `v1.6.1`
 - [ ] Apache Kafka version: `2.6.0`
 - [ ] Operating system: `binaries are run through an analysis tool.`

**Comment by taojwmware** [comment]
<img width="986" alt="Screen Shot 2021-03-09 at 9 06 56 PM" src="https://user-images.githubusercontent.com/25186652/110570538-a9ee5100-811b-11eb-9043-f87803c90274.png">
<img width="989" alt="Screen Shot 2021-03-09 at 9 07 27 PM" src="https://user-images.githubusercontent.com/25186652/110570540-ab1f7e00-811b-11eb-890d-439fe048f6e1.png">

**Comment by edenhill** [comment]
Yes, this is a problem, one that used to be solved when distro packages was the name of the game, but now with each language having its own distribution model and there is no clear solution for external dependencies we need to bundle other software, such as OpenSSL, to make the user experience acceptable.

I'll investigate why the alpine library is still on 1.0.2, should be on 1.1.1, and the next librdkafka release will use the latest OpenSSL 1.1.1 version.

**Comment by edenhill** [comment]
Alpine 3.8, which we use to build the alpine libs, is still on 1.0.2. I'll upgrade to 3.12 which has up to date openssl.

**Comment by edenhill** [comment]
https://github.com/edenhill/librdkafka/pull/3288/commits/075e0f297ec5d46e01ee458e9698e2f7d1ceb576

**Comment by edenhill** [comment]
https://github.com/edenhill/librdkafka/pull/3288
